### PR TITLE
Add optional TorchJD training with sparse DDP geometry

### PR DESCRIPTION
**Pull request by granthamtaylor**

### CODE REVIEW

File: docs/guides/jacobian-descent.qmd [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+model parameters owned by the optimizer. It forms their pairwise inner
+products and asks [TorchJD UPGrad](https://torchjd.org/stable/docs/aggregation/upgrad/)
+for weights. Unit preferences preserve the ordinary summed-gradient scale
+when no projection is needed. The final weighted scalar goes through
+Lightning's normal backward, optimizer, clipping, scheduler and precision
+handling. The logged total loss remains the original sum so its meaning does
+not change with UPGrad weights.
+
+UPGrad resolves conflicts in the raw parameter gradients. AdamW's momentum,
+preconditioning and weight decay still affect the eventual update. Neither
```
Comment (Reid-Taylor) [inline docs/guides/jacobian-descent.qmd]: UPGrad does provide a theoretical proof of non-degradation of all target values for a sufficiently small learning rate. Could link to their paper here, if you're interested in doing so.
